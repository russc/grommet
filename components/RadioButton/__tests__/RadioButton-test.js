"use strict";

var _react = _interopRequireDefault(require("react"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

require("jest-styled-components");

var _Grommet = require("../../Grommet");

var _ = require("..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

test('RadioButton renders', function () {
  var component = _reactTestRenderer.default.create(_react.default.createElement(_Grommet.Grommet, null, _react.default.createElement(_.RadioButton, null), _react.default.createElement(_.RadioButton, {
    id: "test id",
    name: "test name"
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('RadioButton label renders', function () {
  var component = _reactTestRenderer.default.create(_react.default.createElement(_Grommet.Grommet, null, _react.default.createElement(_.RadioButton, {
    label: "test label"
  }), _react.default.createElement(_.RadioButton, {
    label: _react.default.createElement("div", null, "test label")
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('RadioButton checked renders', function () {
  var component = _reactTestRenderer.default.create(_react.default.createElement(_Grommet.Grommet, null, _react.default.createElement(_.RadioButton, {
    checked: true
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('RadioButton disabled renders', function () {
  var component = _reactTestRenderer.default.create(_react.default.createElement(_Grommet.Grommet, null, _react.default.createElement(_.RadioButton, {
    disabled: true
  }), _react.default.createElement(_.RadioButton, {
    disabled: true,
    checked: true
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});