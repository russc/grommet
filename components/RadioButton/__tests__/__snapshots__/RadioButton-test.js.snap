// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton checked renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 > div:first-child {
  margin-right: 12px;
}

.c1:hover input:not([disabled]) + div {
  border-color: #000000;
}

.c3 {
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  margin: 0;
  z-index: 1;
}

.c3:focus + div,
.c3:focus + span {
  border-color: #FD6FFF;
  box-shadow: 0 0 2px 2px #FD6FFF;
}

.c3:focus + div > circle,
.c3:focus + span > circle,
.c3:focus + div > ellipse,
.c3:focus + span > ellipse,
.c3:focus + div > line,
.c3:focus + span > line,
.c3:focus + div > path,
.c3:focus + span > path,
.c3:focus + div > polygon,
.c3:focus + span > polygon,
.c3:focus + div > polyline,
.c3:focus + span > polyline,
.c3:focus + div > rect,
.c3:focus + span > rect {
  outline: #FD6FFF solid 2px;
}

.c3:checked + div {
  border-color: #7D4CDB;
}

.c3:checked + div > svg {
  display: block;
}

.c3:checked + span > span {
  left: 24px;
  background: #7D4CDB;
}

.c4 {
  box-sizing: border-box;
  position: relative;
  top: -1px;
  display: inline-block;
  width: 24px;
  height: 24px;
  vertical-align: middle;
  background: inherit;
  border: 2px solid;
  border-color: rgba(0,0,0,0.15);
  border-radius: 100%;
}

.c4 > svg {
  box-sizing: border-box;
  position: absolute;
  top: -2px;
  left: -2px;
  display: none;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
}

.c2 {
  position: relative;
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    <div
      className="c2"
    >
      <input
        checked={true}
        className="c3"
        type="radio"
      />
      <div
        className="c4"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <circle
            cx={12}
            cy={12}
            r={6}
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`RadioButton disabled renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  opacity: 0.5;
  cursor: default;
}

.c1 > div:first-child {
  margin-right: 12px;
}

.c1:hover input:not([disabled]) + div {
  border-color: #000000;
}

.c3 {
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  margin: 0;
  z-index: 1;
}

.c3:focus + div,
.c3:focus + span {
  border-color: #FD6FFF;
  box-shadow: 0 0 2px 2px #FD6FFF;
}

.c3:focus + div > circle,
.c3:focus + span > circle,
.c3:focus + div > ellipse,
.c3:focus + span > ellipse,
.c3:focus + div > line,
.c3:focus + span > line,
.c3:focus + div > path,
.c3:focus + span > path,
.c3:focus + div > polygon,
.c3:focus + span > polygon,
.c3:focus + div > polyline,
.c3:focus + span > polyline,
.c3:focus + div > rect,
.c3:focus + span > rect {
  outline: #FD6FFF solid 2px;
}

.c3:checked + div {
  border-color: #7D4CDB;
}

.c3:checked + div > svg {
  display: block;
}

.c3:checked + span > span {
  left: 24px;
  background: #7D4CDB;
}

.c4 {
  box-sizing: border-box;
  position: relative;
  top: -1px;
  display: inline-block;
  width: 24px;
  height: 24px;
  vertical-align: middle;
  background: inherit;
  border: 2px solid;
  border-color: rgba(0,0,0,0.15);
  border-radius: 100%;
}

.c4 > svg {
  box-sizing: border-box;
  position: absolute;
  top: -2px;
  left: -2px;
  display: none;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
}

.c2 {
  position: relative;
}

<div
  className="c0"
>
  <label
    className="c1"
    disabled={true}
  >
    <div
      className="c2"
    >
      <input
        className="c3"
        disabled={true}
        type="radio"
      />
      <div
        className="c4"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <circle
            cx={12}
            cy={12}
            r={6}
          />
        </svg>
      </div>
    </div>
  </label>
  <label
    className="c1"
    disabled={true}
  >
    <div
      className="c2"
    >
      <input
        checked={true}
        className="c3"
        disabled={true}
        type="radio"
      />
      <div
        className="c4"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <circle
            cx={12}
            cy={12}
            r={6}
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`RadioButton label renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 > div:first-child {
  margin-right: 12px;
}

.c1:hover input:not([disabled]) + div {
  border-color: #000000;
}

.c3 {
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  margin: 0;
  z-index: 1;
}

.c3:focus + div,
.c3:focus + span {
  border-color: #FD6FFF;
  box-shadow: 0 0 2px 2px #FD6FFF;
}

.c3:focus + div > circle,
.c3:focus + span > circle,
.c3:focus + div > ellipse,
.c3:focus + span > ellipse,
.c3:focus + div > line,
.c3:focus + span > line,
.c3:focus + div > path,
.c3:focus + span > path,
.c3:focus + div > polygon,
.c3:focus + span > polygon,
.c3:focus + div > polyline,
.c3:focus + span > polyline,
.c3:focus + div > rect,
.c3:focus + span > rect {
  outline: #FD6FFF solid 2px;
}

.c3:checked + div {
  border-color: #7D4CDB;
}

.c3:checked + div > svg {
  display: block;
}

.c3:checked + span > span {
  left: 24px;
  background: #7D4CDB;
}

.c4 {
  box-sizing: border-box;
  position: relative;
  top: -1px;
  display: inline-block;
  width: 24px;
  height: 24px;
  vertical-align: middle;
  background: inherit;
  border: 2px solid;
  border-color: rgba(0,0,0,0.15);
  border-radius: 100%;
}

.c4 > svg {
  box-sizing: border-box;
  position: absolute;
  top: -2px;
  left: -2px;
  display: none;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
}

.c2 {
  position: relative;
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    <div
      className="c2"
    >
      <input
        className="c3"
        type="radio"
      />
      <div
        className="c4"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <circle
            cx={12}
            cy={12}
            r={6}
          />
        </svg>
      </div>
    </div>
    <div>
      test label
    </div>
  </label>
  <label
    className="c1"
  >
    <div
      className="c2"
    >
      <input
        className="c3"
        type="radio"
      />
      <div
        className="c4"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <circle
            cx={12}
            cy={12}
            r={6}
          />
        </svg>
      </div>
    </div>
    <div>
      test label
    </div>
  </label>
</div>
`;

exports[`RadioButton renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 > div:first-child {
  margin-right: 12px;
}

.c1:hover input:not([disabled]) + div {
  border-color: #000000;
}

.c3 {
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  margin: 0;
  z-index: 1;
}

.c3:focus + div,
.c3:focus + span {
  border-color: #FD6FFF;
  box-shadow: 0 0 2px 2px #FD6FFF;
}

.c3:focus + div > circle,
.c3:focus + span > circle,
.c3:focus + div > ellipse,
.c3:focus + span > ellipse,
.c3:focus + div > line,
.c3:focus + span > line,
.c3:focus + div > path,
.c3:focus + span > path,
.c3:focus + div > polygon,
.c3:focus + span > polygon,
.c3:focus + div > polyline,
.c3:focus + span > polyline,
.c3:focus + div > rect,
.c3:focus + span > rect {
  outline: #FD6FFF solid 2px;
}

.c3:checked + div {
  border-color: #7D4CDB;
}

.c3:checked + div > svg {
  display: block;
}

.c3:checked + span > span {
  left: 24px;
  background: #7D4CDB;
}

.c4 {
  box-sizing: border-box;
  position: relative;
  top: -1px;
  display: inline-block;
  width: 24px;
  height: 24px;
  vertical-align: middle;
  background: inherit;
  border: 2px solid;
  border-color: rgba(0,0,0,0.15);
  border-radius: 100%;
}

.c4 > svg {
  box-sizing: border-box;
  position: absolute;
  top: -2px;
  left: -2px;
  display: none;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
}

.c2 {
  position: relative;
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    <div
      className="c2"
    >
      <input
        className="c3"
        type="radio"
      />
      <div
        className="c4"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <circle
            cx={12}
            cy={12}
            r={6}
          />
        </svg>
      </div>
    </div>
  </label>
  <label
    className="c1"
    htmlFor="test id"
  >
    <div
      className="c2"
    >
      <input
        className="c3"
        id="test id"
        name="test name"
        type="radio"
      />
      <div
        className="c4"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <circle
            cx={12}
            cy={12}
            r={6}
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;
